adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -259,64 +262,65 @@ WordLoop:
 }
 
 func (b *BM25Searcher) getTopKObjects(topKHeap *priorityqueue.Queue[any],
-	results terms, indices []map[uint64]int, additionalExplanations bool,
+	results terms, indices []map[uint64]int, additionalExplanations bool, additional additional.Properties,
 ) ([]*storobj.Object, []float32, error) {
 	objectsBucket := b.store.Bucket(helpers.ObjectsBucketLSM)
 	if objectsBucket == nil {
 		return nil, nil, errors.Errorf("objects bucket not found")
 	}
 
-	objects := make([]*storobj.Object, 0, topKHeap.Len())
 	scores := make([]float32, 0, topKHeap.Len())
-
-	buf := make([]byte, 8)
+	ids := make([]uint64, 0, topKHeap.Len())
 	for topKHeap.Len() > 0 {
 		res := topKHeap.Pop()
-		binary.LittleEndian.PutUint64(buf, res.ID)
-		objectByte, err := objectsBucket.GetBySecondary(0, buf)
-		if err != nil {
-			return nil, nil, err
-		}
+		ids = append(ids, res.ID)
+		scores = append(scores, res.Dist)
+	}
 
-		// If there is a crash and WAL recovery, the inverted index may have objects that are not in the objects bucket.
-		// This is an issue that needs to be fixed, but for now we need to reduce the huge amount of log messages that
-		// are generated by this issue. Logging the first time we encounter a missing object in a query still resulted
-		// in a huge amount of log messages and it will happen on all queries, so we not log at all for now.
-		// The user has already been alerted about ppossible data loss when the WAL recovery happened.
-		// TODO: consider deleting these entries from the inverted index and alerting the user
-		if len(objectByte) == 0 {
-			continue
-		}
+	objs, err := storobj.ObjectsByDocID(objectsBucket, ids, additional, b.logger)
+	if err != nil {
+		return objs, nil, errors.Errorf("objects loading")
+	}
 
-		obj, err := storobj.FromBinary(objectByte)
-		if err != nil {
-			return nil, nil, err
+	// handle case that an object was removed
+	if len(objs) != len(scores) {
+		j := 0
+		for i := range scores {
+			if j >= len(objs) {
+				break
+			}
+			if objs[j].DocID != ids[i] {
+				continue
+			}
+			scores[j] = scores[i]
+			j++
 		}
+		scores = scores[:j]
+	}
 
-		if additionalExplanations {
+	if additionalExplanations {
+		for k := range objs {
 			// add score explanation
-			if obj.AdditionalProperties() == nil {
-				obj.Object.Additional = make(map[string]interface{})
+			if objs[k].AdditionalProperties() == nil {
+				objs[k].Object.Additional = make(map[string]interface{})
 			}
 			for j, result := range results {
-				if termIndex, ok := indices[j][res.ID]; ok {
+				if termIndex, ok := indices[j][ids[k]]; ok {
```

Comment:
`ids[k]` might not be matching the desired object if some object was deleted? because `ids` holds all the objects but `objs` the non-deleted ones?

Reply:
```for k, obj := range objs {``` could be used